frameit/lib/frameit/editor.rb

```diff
@@ -117,9 +117,19 @@ def complex_framing
         # Decrease the size of the framed screenshot to fit into the defined padding + background
         frame_width = background.width - horizontal_frame_padding * 2
         frame_height = background.height - top_space_above_device - vertical_frame_padding
-        @image.resize "#{frame_width}x"
-        if @image.height > frame_height
-          @image.resize "x#{frame_height.to_i}"
+
+        if fetch_config['show_complete_frame']
+          # calculate the final size of the screenshot to resize in one go
+          # it may be limited either by the width or height of the frame
+          image_aspect_ratio = @image.width.to_f / @image.height.to_f
+          image_width = [frame_width, @image.width].min
+          image_height = [frame_height, image_width / image_aspect_ratio].min
+          image_width = image_height * image_aspect_ratio
```

Comment:
We want to calculate the final frame for a given outer/limiting frame and an aspect ratio. Now we have two cases:
1) The `frame_width` is the limiting factor. Then `image_width / image_aspect_ratio` is < `frame_height`. In this case, line 127 is not needed.
2) The `frame_height` is the limiting factor. Then `image_height` == `frame_height` and `image_width / image_aspect_ratio` is larger than that. In this case, `image_width` must be smaller than `frame_width` by calculating it back from the `image_height` and the aspect ratio.

Does that make sense?